File: backend/app/schemas/alert.py
from datetime import datetime

from pydantic import BaseModel


class AlertRead(BaseModel):
    id: int
    target_id: int
    target_name: str | None = None
    target_url: str | None = None
    level: str
    title: str
    content: str
    is_read: bool
    created_at: datetime


class AlertListResponse(BaseModel):
    items: list[AlertRead]
    total: int

File: frontend/types/alert.ts
export type AlertLevel = "info" | "success" | "warning" | "error";

export type AlertRecord = {
  id: number;
  target_id: number;
  target_name?: string | null;
  target_url?: string | null;
  level: AlertLevel;
  title: string;
  content: string;
  is_read: boolean;
  created_at: string;
};

export type AlertListResponse = {
  items: AlertRecord[];
  total: number;
};
